runner/browserrunner.ts

```diff
@@ -0,0 +1,211 @@
+          console.log(data.value.message);
+          if (data.value && data.value.message && /Failed to connect to SafariDriver/i.test(data.value.message)) {
+            error = 'Until Selenium\'s SafariDriver supports Safari 6.2+, 7.1+, & 8.0+, you must\n' +
+                    'manually install it. Follow the steps at:\n' +
+                    'https://github.com/SeleniumHQ/selenium/wiki/SafariDriver#getting-started';
+          }
+        } catch (error) {
+          // Show the original error.
+        }
+      }
+      // END TEMPORARY CHECK
+      this.done(error.data || error);
+    } else {
+      this.sessionId = sessionId;
+      this.startTest();
+      this.extendTimeout();
+    }
+  }
+
+  startTest() {
+    const host  = 'http://' + this.options.webserver.hostname + ':' + this.options.webserver.port;
+    const path  = this.options.webserver.webRunnerPath;
+    const extra = (path.indexOf('?') === -1 ? '?' : '&') + 'cli_browser_id=' + this.def.id;
+    this.browser.get(host + path + extra, (error) => {
+      if (error) {
+        this.done(error.data || error);
+      } else {
+        this.extendTimeout();
+      }
+    });
+  }
+
+  onEvent(event: string, data: any) {
+    this.extendTimeout();
+
+    if (event === 'browser-start') {
+      // Always assign, to handle re-runs (no browser-init).
+      this.stats = {
+        status:  'running',
+        passing: 0,
+        pending: 0,
+        failing: 0,
+      };
+    } else if (event === 'test-end') {
+      this.stats[data.state] = this.stats[data.state] + 1;
+    }
+
+    if (event === 'browser-end') {
+      this.done(data);
+    } else {
+      this.emitter.emit(event, this.def, data, this.stats, this.browser);
+    }
+  }
+
+  done(error: any) {
+    // No quitting for you!
+    if (this.options.persistent) return;
+
+    if (this.timeoutId) clearTimeout(this.timeoutId);
+    // Don't double-quit.
+    if (!this.browser) return;
+    const browser = this.browser;
+    this.browser = null;
+
+    this.stats.status = error ? 'error' : 'complete';
+    if (!error && this.stats.failing > 0) {
+      error = this.stats.failing + ' failed tests';
+    }
+
+    this.emitter.emit('browser-end', this.def, error, this.stats, this.sessionId, browser);
+
+    // Nothing to quit.
+    if (!this.sessionId) {
+      return this.doneCallback(error, this);
+    }
+
+    browser.quit((quitError) => {
+      if (quitError) {
+        this.emitter.emit('log:warn', this.def, 'Failed to quit:', quitError.data || quitError);
+      }
+      this.doneCallback(error, this);
+    });
+  }
+
+  extendTimeout() {
+    if (this.options.persistent) return;
+    if (this.timeoutId) clearTimeout(this.timeoutId);
+    this.timeoutId = setTimeout(() => {
+      this.done('Timed out');
+    }, this.timeout);
+  }
+
+  quit() {
+    this.done('quit was called');
+  }
+
+  //HACK
+  static BrowserRunner = BrowserRunner;
+}
+
+module.exports = BrowserRunner;
```

Comment:
Easy, but it would be a breaking change for anyone who imported any of the non-ES-compliant modules.

I'm currently using a midway hack everywhere here by exposing the default value as a property on itself. That way both we support both the new and the old patterns.

I was thinking we'd do one back-compat release with typescript and then a breaking release with a bunch of stuff improved (like callbacks -> promises for example).